### PR TITLE
[stdlib][NFC] Warning cleanup

### PR DESCRIPTION
**Pull request by airspeedswift**
Couple of uses of `CountableRange`, a near-miss, remove a deprecation warning that won't be missed, and mark `_SwiftNativeNSArrayWithContiguousStorage.init` `@nonobjc`

### CODE REVIEW

File: stdlib/public/core/SwiftNativeNSArray.swift [hunk 1]
```diff
@@ -48,7 +48,7 @@ internal class _SwiftNativeNSArrayWithContiguousStorage
 
   @_inlineable // FIXME(sil-serialize-all)
   @_versioned // FIXME(sil-serialize-all)
-  internal override init() {}
+  @nonobjc internal override init() {}
```
Comment (airspeedswift) [inline stdlib/public/core/SwiftNativeNSArray.swift]: I think this is right, since these only get created from within Swift.

**Comment by airspeedswift** [comment]
@swift-ci please test

**Comment by lancep** [comment]
LGTM!

**Review by moiseev — APPROVED** [review]
:shipit:

**Pull request merged**